pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -3386,6 +3387,222 @@ var _ = Context("NodePool", func() {
+				ObjectMeta: metav1.ObjectMeta{Name: pod.Name, Namespace: pod.Namespace},
+			})).To(Succeed())
+			ExpectEvicted(ctx, env.Client, pod)
+			ExpectExists(ctx, env.Client, pod)
+
+			// Trigger a provisioning loop and expect that we don't create more nodes since we don't consider
+			// generic terminating pods
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov)
+
+			// We shouldn't create an additional node here because this is a standard pod
+			nodes := ExpectNodes(ctx, env.Client)
+			Expect(nodes).To(HaveLen(1))
+		})
+		It("should not re-schedule pods from a deleting node when pods are owned by a DaemonSet", func() {
+			ds := test.DaemonSet()
+			ExpectApplied(ctx, env.Client, nodePool, ds)
+
+			pod := test.UnschedulablePod(
+				test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion:         "apps/v1",
+								Kind:               "DaemonSet",
+								Name:               ds.Name,
+								UID:                ds.UID,
+								Controller:         ptr.Bool(true),
+								BlockOwnerDeletion: ptr.Bool(true),
+							},
+						},
+					},
+					ResourceRequirements: v1.ResourceRequirements{
+						Requests: map[v1.ResourceName]resource.Quantity{
+							v1.ResourceMemory: resource.MustParse("100M"),
+						},
+					},
+				},
+			)
+			nodeClaim, node := test.NodeClaimAndNode(v1beta1.NodeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1beta1.NodePoolLabelKey:     nodePool.Name,
+						v1.LabelInstanceTypeStable:   "small-instance-type",
+						v1beta1.CapacityTypeLabelKey: v1beta1.CapacityTypeOnDemand,
+						v1.LabelTopologyZone:         "test-zone-1a",
+					},
+				},
+				Status: v1beta1.NodeClaimStatus{
+					Allocatable: map[v1.ResourceName]resource.Quantity{v1.ResourceCPU: resource.MustParse("32")},
+				},
+			})
+			ExpectApplied(ctx, env.Client, nodeClaim, node, pod)
+
+			ExpectManualBinding(ctx, env.Client, pod, node)
+
+			// Mark for deletion so that we consider all pods on this node for reschedulability
+			cluster.MarkForDeletion(node.Spec.ProviderID)
+
+			// Trigger an eviction to set the deletion timestamp but not delete the pod
+			Expect(env.KubernetesInterface.PolicyV1().Evictions(pod.Namespace).Evict(ctx, &policyv1.Eviction{
+				ObjectMeta: metav1.ObjectMeta{Name: pod.Name, Namespace: pod.Namespace},
+			})).To(Succeed())
+			ExpectEvicted(ctx, env.Client, pod)
+			ExpectExists(ctx, env.Client, pod)
+
+			// Trigger a provisioning loop and expect that we don't create more nodes since we don't consider
+			// generic terminating pods
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov)
+
+			// We shouldn't create an additional node here because this is a standard pod
+			nodes := ExpectNodes(ctx, env.Client)
+			Expect(nodes).To(HaveLen(1))
+		})
+		It("should not reschedule pods from a deleting node when pods are not active and they are owned by a ReplicaSet", func() {
```

Comment:
Should we actually be rescheduling pods from a deleting node if the previous replica gets force terminated before the new replica actually goes active? I think right now we'd undershoot here, by not including this pod.

Reply:
> I think right now we'd undershoot here, by not including this pod

I'm not sure how we would know this. In the general case the new replicas get created immediately (or as quickly as they can be). Force deleting is a rare occurrence and, in the general case for replicasets and deployments, you are going to get a pending pod spun up as soon as the old pod goes into a terminating state.

Reply:
So the idea is potentially undershooting, but possibly getting pulled into the provisioning loop anyway through the batching, is that right?

Reply:
> but possibly getting pulled into the provisioning loop anyway through the batching, is that right

I don't think we will generally undershoot. In the worst-case, yes we may undershoot, but this should be rare. Realistically, this is better than the alternative where we overshoot and then have to pull it back through consolidation. Or worse -- they haven't enabled consolidation and then they are just overpaying.